CRM/Upgrade/Incremental/php/FiveFortyTwo.php

```diff
@@ -43,10 +43,14 @@ public function setPreUpgradeMessage(&$preUpgradeMessage, $rev, $currentVer = NU
    *   an intermediate version; note that setPostUpgradeMessage is called repeatedly with different $revs.
    */
   public function setPostUpgradeMessage(&$postUpgradeMessage, $rev) {
-    // Example: Generate a post-upgrade message.
-    // if ($rev == '5.12.34') {
-    //   $postUpgradeMessage .= '<br /><br />' . ts("By default, CiviCRM now disables the ability to import directly from SQL. To use this feature, you must explicitly grant permission 'import SQL datasource'.");
-    // }
+    $templateUpgrader = new CRM_Upgrade_Incremental_MessageTemplates($rev);
+    $upgradeMessage[] = $templateUpgrader->getMessageTemplateWarning('contribution_invoice_receipt', '$contributionMode', 'no longer available / relevant');
+    $upgradeMessage[] = $templateUpgrader->getMessageTemplateWarning('contribution_invoice_receipt', '$first_name', 'contact.first_name');
+    $upgradeMessage[] = $templateUpgrader->getMessageTemplateWarning('contribution_invoice_receipt', '$last_name', 'contact.last_name');
+    $upgradeMessage[] = $templateUpgrader->getMessageTemplateWarning('contribution_invoice_receipt', '$displayName', 'contact.display_name');
+    if (!empty($upgradeMessage)) {
```

Comment:
@totten so if I just remove this bit is it right ? (Message wise - the face it is checking before rather than after is a separate & maybe not that solvable issue)

Reply:
Yeah, that emptiness check doesn't seem right. Arrays like `[NULL,NULL,NULL]` and `['','','']` are not `empty()`, but they are devoid of meaningful content. Maybe something like this...

```php
$upgradeMessage = array_filter($upgradeMessage, function($v) { return !empty($v); });
```

... would trim the array down to only show meaningful content.